core/json/src/main/scala/net/liftweb/json/Extraction.scala

```diff
@@ -319,14 +331,54 @@ object Extraction {
       }
     }
 
+    def newTuple(root: JValue, mappings: List[Mapping]): Any = {
+      root match {
+        case JArray(items) if items.length >= 1 && items.length <= 22 =>
+          val builtItems: Seq[Object] = items.zip(mappings).map({
+            case (i,m) =>
+              build(i, m).asInstanceOf[Object]
+          })
+          val tupleClass = Class.forName("scala.Tuple" + items.length)
+
+          val typedTupleConstructor = tupleClass.getConstructors()(0)
+          typedTupleConstructor.newInstance(builtItems: _*)
+
+        case JObject(items) if items.forall(_.name.startsWith("_")) =>
+          newTuple(JArray(items.map(_.value)), mappings)
```

Comment:
I have a back-compat test somewhere, I think, that suggests otherwise. If so, I'll point to it here. If not, I'll take a closer look at it.